### PR TITLE
test: proxy tests

### PR DESCRIPTION
**Pull request by ishanarya0**

### CODE REVIEW

**Comment by ravisuhag** [comment]
@krtkvrm @ishanarya0 is this PR valid after 0.5 merge?

**Comment by krtkvrm** [comment]
we will continuing this in v0.5.1

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -39,4 +39,47 @@ jobs:
...
+        with:
+          go-version: '1.18'
+      - name: install dependencies
+        run: go mod tidy
+      - name: install spicedb binary
+        uses: authzed/action-spicedb@v1
+      - name: run proxy tests
+        env:
+          POSTGRES_PASSWORD: postgres
+        run: make e2e-smoke-test
```
Comment (krtkvrm) [inline .github/workflows/test.yml:85]: add line at end

File: test/e2e_test/smoke/proxy_test.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+		logger.Fatal("failed to serve proxies", err)
+		return
+	}
+}
+
+func (s *EndToEndProxySmokeTestSuite) TearDownTest() {
+}
+
+func (s *EndToEndProxySmokeTestSuite) TestProxyToEchoServer() {
+	s.Run("1. should be able to proxy to an echo server", func() {
```
Comment (krtkvrm) [inline test/e2e_test/smoke/proxy_test.go:189]: can remove numbers

**Review by krtkvrm — CHANGES_REQUESTED** [review]

**Pull request merged**